novuhq/api-docs
# feat: add Algolia search, improvements to the mobile version and navigation

**Describe what changes this pull request brings**
This pull request adds the `Algolia search` for pages.

**Describe any additional changes**

improvement for the mobile version:
<details>
<summary>Before:</summary>

https://user-images.githubusercontent.com/17677196/186920497-7c33fbea-9ff3-42f8-8a8a-516683ed4990.mov


</details>
<details>
<summary>After:</summary>

https://user-images.githubusercontent.com/17677196/186920450-bd96d5dd-c93f-4a90-9a2e-6715f21eb105.mov


</details>

fixing the active menu item when the page initially loads by anchor + scrolling to the active item:
<details>
<summary>Before:</summary>

https://user-images.githubusercontent.com/17677196/186920365-da434600-4862-40d4-aa0d-01e8c471dc0b.mov


</details>
<details>
<summary>After:</summary>

https://user-images.githubusercontent.com/17677196/186920285-4571c72a-6b44-481c-b959-c0f093f572ed.mov


</details>

code optimization and refactoring

**Steps to test**
1. Open [the page](https://novuapidocs-algoliasearch.gtsb.io/api/overview/)
2. Confirm that everything looks and works as expected
